youtube_dl/extractor/yle.py

```diff
@@ -0,0 +1,278 @@
+            media_response = self._download_json(media_url, media_id)
+            media_infos.extend(media_response['data']['media'][protocol])
+
+        has_subtitles = [i for i in media_infos if i.get('subtitles')]
+
+        formats = []
+        for fmt in media_infos:
+            if has_subtitles and 'hardsubtitle' in fmt:
+                continue
+
+            url = self._decrypt_data(fmt['url']).decode('utf-8')
+            sub_list = fmt.get('subtitles', [])
+
+            extracted_subtitles = self.extract_subtitles(sub_list)
+            if subtitles:
+                if subtitles != extracted_subtitles:
+                    raise ExtractorError('Different formats return different subtitle urls')
+            else:
+                subtitles = extracted_subtitles
+            if fmt['protocol'] == 'HDS':
+                sep = '&' if '?' in url else '?'
+                url += sep + 'hdcore=3.3.0&plugin=flowplayer-3.3.0.0'
+                try:
+                    formats.extend(self._extract_f4m_formats(url, media_id, f4m_id='hds'))
+                except ExtractorError as e:
+                    if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                        continue
+                    raise
+                continue
+
+            abr = fmt.get('audioBitrateKbps', 0)
+            vbr = fmt.get('videoBitrateKbps', 0)
+            tbr = vbr + abr
+
+            formats.append({
+                'format_id': 'rtmp-%d' % tbr if tbr else 'rtmp',
+                'url': url,
+                'ext': 'flv',
+                'abr': abr if abr else None,
+                'vbr': vbr if vbr else None,
+                'tbr': tbr if tbr else None,
+                'height': fmt.get('height'),
+                'width': fmt.get('width'),
+                'rtmp_live': fmt.get('live'),
+                'protocol': 'rtmp',
+            })
+
+        return formats, subtitles
+
+    def _get_subtitles(self, sub_list):
+        subtitles = {}
+        for sub in sub_list:
+            subtitles[sub['lang']] = {
+                'ext': 'srt',
+                'url': sub['uri'],
+            }
+        return subtitles
+
+    def _decrypt_data(self, data):
+        data = bytes_to_intlist(base64.b64decode(data))
+        key = bytes_to_intlist(self._AES_KEY.encode('utf-8'))
+
+        iv = data[:BLOCK_SIZE_BYTES]
+        cipher = data[BLOCK_SIZE_BYTES:]
+        decrypted_data = aes_cfb_decrypt(cipher, key, iv)
+
+        plaintext = intlist_to_bytes(decrypted_data)
+
+        return plaintext
+
+
+class YLEAreenaLiveTVIE(YLEAreenaIE):
+    _VALID_URL = r'^http://(?P<host>areena|arenan)\.yle\.fi/tv/(?:suorat|direkt)/(?P<id>[\w-]+).*'
+    _PROGRAM_URL = 'http://player.yle.fi/api/v1/services.jsonp?id={pid}&region=fi'
+    _PROGRAM_KEY = 'service'
+    _EVENT_KEY = 'outlet'
+
+    _TEST = {
+        'url': 'http://areena.yle.fi/tv/suorat/yle-tv1',
+        'info_dict': {
+            'title': 'Yle TV1',
```

Comment:
For lives you should build `title` with `self._live_title` and set `is_live` to `True`.